sycl/test-e2e/Graph/graph_common.hpp

```diff
@@ -14,6 +14,16 @@ namespace exp_ext = sycl::ext::oneapi::experimental;
 // Make tests less verbose by using sycl namespace.
 using namespace sycl;
 
+// Macro for wrapping depends on calls when add_node is used so they are not
+// used in the explicit API
+#ifdef GRAPH_E2E_RECORD_REPLAY
+#define DEPENDS_ON(CGH, EVENTS) CGH->depends_on(EVENTS)
+#else
+#define DEPENDS_ON(CGH, EVENTS)                                                \
+  do {                                                                         \
+  } while (0)
```

Comment:
```suggestion
#define DEPENDS_ON(CGH, EVENTS) ((void)0)
```
This might look better. But do we really want to drop dependencies in explicit mode?

Reply:
Dependencies for the explicit API are passed to the generic `add_node()` function so they can be used only with the explicit mode. This macro is not intended for use outside of that function. In explicit mode we don't have events returned from adding nodes to the graph, so we cannot call `handler::depends_on()`. The purpose of this macro is that so it can still be used with the recording API where we do have events and rely on them for establishing dependencies.

Reply:
As an update to this I've ended up replacing this macro with a template function since handling more than one event was problematic, but it has the same function and in explicit mode the calls will still be a no-op.